apis/projectcontour/v1/httpproxy.go

```diff
@@ -314,6 +314,10 @@ type TimeoutPolicy struct {
 	Idle string `json:"idle,omitempty"`
 }
 
+// RetryOn is a string type alias with validation to ensure that the value is valid.
+// +kubebuilder:validation:Enum=5xx;gateway-error;reset;connect-failure;retriable-4xx;refused-stream;retriable-status-codes;retriable-headers
```

Comment:
Not sure what the right way to test this validation is.

Reply:
Probably verification by hand would be sufficient (that is, run `make generate`, apply to a cluster, then try add add things that don't match this list).

Reply:
You can also test validation with integration tests, e.g. https://github.com/projectcontour/contour/blob/master/_integration/testsuite/httpproxy/001-required-field-validation.yaml